packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -104,41 +104,124 @@ export class AnchorSet {
 
     /**
      * Updates paths for a range move (including re-parenting path items and updating indexes).
+     * @param count - number of siblings to to insert/delete/move.
+     * @param src - where the siblings are removed from. If undefined the operation is an insert.
+     * @param dst - where the siblings are moved to. If undefined the operation is a delete.
+     *
+     * TODO:
+     * How should anchors that become invalid, then valid again (ex: into content that was deleted, then undone) work?
+     * Add an API to resurrect them? Store them in special detached fields? Store them in special non-detached fields?
+     *
+     * TODO:
+     * Now should custom anchors work (ex: ones not just tied to a specific Node)?
+     * This design assumes they can be expressed in terms of a Node anchor + some extra stuff,
+     * but we don't have an API for the extra stuff yet.
      */
     public moveChildren(
-        src: UpPath, srcField: FieldKey, start: number, count: number,
-        dst: UpPath, dstField: FieldKey, dstIndex: number): void {
-            const srcPath = this.find(src);
+        count: number,
+        src: undefined | { path: UpPath; field: FieldKey; start: number; },
+        dst: undefined | { path: UpPath; field: FieldKey; start: number; },
+    ): void {
+        assert(
+            src !== undefined || dst !== undefined,
+            "moveChildren is a no-op and should not be called if there is no src or dst",
+        );
+
+        const srcParent = src === undefined ? undefined : this.find(src.path);
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const srcChildren = srcParent?.children?.get(src!.field);
             // Sorted list of PathNodes to move from src to dst.
             let toMove: PathNode[];
-            if (srcPath !== undefined) {
-                // TODO: set toMove
-                // Remove items from src.
+
+        // Update src
+        if (srcChildren !== undefined) {
+            let numberBeforeMove = 0;
+            let numberToMove = 0;
+            for (let index = 0; index < srcChildren.length; index++) {
+                const element = srcChildren[index];
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                if (element.parentIndex < src!.start) {
+                    numberBeforeMove = index;
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                } else if (element.parentIndex < src!.start + count) {
+                    numberToMove++;
+                } else {
                 // Fix indexes in src after moved items (subtract count).
+                    srcChildren[index].parentIndex -= count;
+                }
+            }
```

Comment:
Consider refactoring into 3 loops without ifs (i.e., take advandtage of the fact that the array is sorted).

Reply:
I guess doing it with while loops could be pretty elegant. I'll consider it.

Reply:
Done